modules/afsql/afsql.c

```diff
@@ -908,9 +991,17 @@ afsql_dd_run_insert_query(AFSqlDestDriver *self, GString *table, LogMessage *msg
   GString *insert_command;
 
   insert_command = afsql_dd_build_insert_command(self, msg, table);
-  gboolean success = afsql_dd_run_query(self, insert_command->str, FALSE, NULL);
+  if(insert_command)
+    {
+      gboolean success = afsql_dd_run_query(self, insert_command->str, FALSE, NULL);
+      return success;
+    }
+  else
+    {
+      msg_error("INSERT statement dropped as drop is enabled and value isn't okay");
+    }
   g_string_free(insert_command, TRUE);
-  return success;
+  return FALSE;
```

Comment:
This aborts the plugin when insert_command is not set.

`g_string_free()` should only be called when `insert_command` is non-NULL.